Code/Framework/AzCore/AzCore/IO/Streamer/RequestPath.cpp

```diff
@@ -37,44 +37,28 @@ namespace AZ
 
         bool RequestPath::operator==(const RequestPath& rhs) const
         {
-            if (m_path.empty() || rhs.m_path.empty())
-            {
-                return m_path.empty() == rhs.m_path.empty();
-            }
-
-            ResolvePath();
-            rhs.ResolvePath();
-            if (m_absolutePathHash != rhs.m_absolutePathHash)
-            {
-                return false;
-            }
-            else
-            {
-                return m_path == rhs.m_path;
-            }
+            return m_absolutePathHash != rhs.m_absolutePathHash ? false : (m_path == rhs.m_path);
```

Comment:
Instead of using a ternary expression "and" `&&` condition can be used

```suggestion
            return m_absolutePathHash == rhs.m_absolutePathHash  && m_path == rhs.m_path;
```

Reply:
Good call, simplified it further.